docs/CREDENTIALS.md

```diff
@@ -0,0 +1,388 @@
+---
+title: Credentials
+category: Concepts
+layout: default
+SPDX-License-Identifier: LGPL-2.1-or-later
+---
+
+# System and Service Credentials
+
+The `systemd` service manager supports a "credential" concept for securely
+acquiring and passing credential data to systems and services. The precise
+nature of the credential data is up to applications, but the concept is
+intended to provide systems and services with potentially security sensitive
+cryptographic keys, certificates, passwords, identity information and similar
+types of information. It may also be used as generic infrastructure for
+parameterizing systems and services.
+
+Traditionally, data of this nature has often been provided to services via
+environment variables (which is problematic because by default it is inherited
+down the process tree, has size limitations, and issues with binary data) or
+simple, unencrypted files on disk. `systemd`'s system and service credentials
+are supposed to provide a better alternative for this purpose. Specifically,
+the following features are provided:
+
+1. Service credentials are acquired at moment of service activation, and
+   released on service deactivation. They are immutable during the service
+   runtime.
+
+2. Service credentials are accessible to service code as regular files, the
+   path to access them is derived from the environment variable
+   `$CREDENTIALS_DIRECTORY`.
+
+3. Access to credentials is restricted to the service's user. Unlike
+   environment variables the credential data is not propagated down the process
+   tree. Instead each time a credential is accessed an access check is enforced
+   by the kernel. If the service is using file system namespacing the loaded
+   credential data is invisble to any other services.
+
+4. Service credentials may be acquired from files on disk, may be specified as
+   literal strings in unit files, may be acquired from another service
+   dynamically via an `AF_UNIX` socket, or inherited from the system
```

Comment:
The many "may be"s made me stumble a bit when reading this. I think this could be shorter like this
```suggestion
4. Service credentials may be acquired from files on disk, specified as
   literal strings in unit files, acquired from another service
   dynamically via an `AF_UNIX` socket, or inherited from the system
```
or, what I think made me pause when reading this, add another "may be"
```suggestion
4. Service credentials may be acquired from files on disk, may be specified as
   literal strings in unit files, may be acquired from another service
   dynamically via an `AF_UNIX` socket, or may be inherited from the system
```